lib/maths/analytics/CBoostedTreeFactory.cc

```diff
@@ -601,6 +625,7 @@ void CBoostedTreeFactory::initializeHyperparametersSetup(core::CDataFrame& frame
     // This needs to be tied to the learn rate to avoid bias.
     hyperparameters.maximumNumberTrees().setToRangeMidpointOr(
         computeMaximumNumberTrees(hyperparameters.eta().value()));
+    hyperparameters.retrainedTreeEta().setToRangeMidpointOr(1.0);
```

Comment:
I would set it to `0.5` instead of `1.0` as a fallback value.

Reply:
The only thing is that this interacts with whether or not you have `predictionChangeCost` non-zero. I was tending to find 1 was good in conjunction with this non-zero. In the multistep train from scratch I agree all your results suggests 1 is a bad choice. How about I set it to 1 if `predictionChangeCost` > 0 and `eta` otherwise?

Reply:
[32a64c3](https://github.com/elastic/ml-cpp/pull/2125/commits/32a64c37b81ec05b0fc5cba2befaff9ca7026849)